### ISSUE TITLE
CLI quits on unknown options

### ISSUE THREAD

**Issue by mmmichl** [angular/angular-cli#10756]
Up to version 1.7 the CLI printed a warning if an unknown parameter is encountered, but continued executing. With 6.0 the CLI quits if an unknown parameter is encountered. We used the former behavior to easily configure e.g. karma with the api endpoint or to change the browser in protractor.

### Versions
```
Angular CLI: 6.0.0
Node: 8.11.1
OS: darwin x64
Angular: 6.0.0
... animations, cli, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.6.0
@angular-devkit/build-angular     0.6.0
@angular-devkit/build-optimizer   0.6.0
@angular-devkit/core              0.6.0
@angular-devkit/schematics        0.6.0
@ngtools/webpack                  6.0.0
@schematics/angular               0.6.0
@schematics/update                0.6.0
rxjs                              6.1.0
typescript                        2.7.2
webpack                           4.6.0
```

### Repro steps
Used examples:
* `ng test --apiUrl http://localhost:18080 --browsers Firefox`
* `e2e --browsers Firefox --no-serve --base-href http://localhost:18080/ui/`

### Observed behavior
With 6.0 the CLI quits, if an unknown parameter is used.

### Desired behavior
Print a warning and ignore the unknown parameter as done in the CLI in `<=1.7`

**Comment by aszechlicki** [comment]
in my opinion this is desired behavior - passing unsupported parameters usually means some error in workflow (e.g. a typo), and ignoring that may lead to unpredictable results of command execution

**Comment by mmmichl** [comment]
In general I agree with you, that quitting on unknown arguments is a good practice. Yet there are a vast variety of situation out there, which is not possible nor feasible to be covered by the CLI. In my case, I need to change a tiny thing for each environment we run the build in. And the only solution I see currently is to make a copy of `karma.config.js` and `protractor.config.js` for each environment, and this is not a solution for me.

Having custom arguments was a very elegant and flexible way to tune/change the execution just fine to cover our needs. I think more users would benefit from such a possibility.

**Comment by jeffgammon** [comment]
I agree with @mmmichl, we use unknown arguments to get custom behavior in both karm.config.js and protractor.config.js for each environment, and creating copies of karma.config.js and protractor.config.js is not an option either. For example, we create suites of tests in karma to quickly run tests for just 1 feature (ng test --suite feature1). And for protractor, we pass an ID to set up the environment for a particular user so as to allow other users running the tests simultaneously in the same environment (ng e2e --id jeff). Command-line parameter was elegant and flexible way to tune execution. Struggling to workaround this limitation with 6.0. Reverting to printing the warning would be great for us.

**Comment by mmmichl** [comment]
@jeffgammon I worked around the issue with environment variables. It's not as easy to handle as before, but it works.
